Tests/GRPCCoreTests/Call/TracingInterceptorTests.swift

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2024, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import GRPCCore
+import GRPCInterceptors
+import Tracing
+import XCTest
+
+@available(macOS 13.0, iOS 16.0, watchOS 9.0, tvOS 16.0, *)
+final class ClientTracingInterceptorTests: TracingTestCase {
+  func testClientInterceptor() async throws {
+    var serviceContext = ServiceContext.topLevel
+    let traceIDString = UUID().uuidString
+    serviceContext.traceID = traceIDString
+    try await ServiceContext.withValue(serviceContext) {
+      let tester = ClientRPCExecutorTestHarness(server: .echo)
```

Comment:
I don't think we actually need this at all for the tests: we can just construct the interceptor and call it directly with a stream request and have the `next` function validate the metadata is injected and return an empty response or something similar.